z3/include/z3_datatype.h

```diff
@@ -0,0 +1,77 @@
+#pragma once
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "datatype.h"
+#include "exceptions.h"
+#include "z3++.h"
+
+namespace smt {
+
+// forward declaration
+class Z3Solver;
+class Z3Datatype;
+class Z3DatatypeConstructorDecl;
+class Z3DatatypeDecl : public AbsDatatypeDecl
```

Comment:
Same question -- I don't see forward declarations in the cvc5 case. Can we try to make the z3 and cvc5 cases uniform? Or is that really needed only for z3?

Reply:
Removed